### PR TITLE
feat(xPro): add EMERITUS_API_KEY and feature flag

### PR DESCRIPTION
**Pull request by asadali145**
### What are the relevant tickets?

A part of https://github.com/mitodl/hq/issues/4032

### Description (What does it do?)

Adds `EMERITUS_API_KEY` setting for xPro Heroku App. Also, it adds the feature flag settings for the `FEATURE_ENABLE_EXTERNAL_COURSE_SYNC`.

### How can this be tested?

- Add the setting locally and run the command `./manage.py sync_external_course_runs --vendor-name emeritus`

### Checklist:
- [x] Update secret values in Vault before merging. The API Key is mentioned in the linked issue.

### CODE REVIEW

File: src/ol_infrastructure/applications/xpro/Pulumi.applications.xpro.Production.yaml [hunk 1]
```diff
@@ -12,6 +12,7 @@ config:
     DRIVE_WEBHOOK_EXPIRATION_MINUTES: 1440
     DRIVE_WEBHOOK_RENEWAL_PERIOD_MINUTES: 720
     FEATURE_ENABLE_ENTERPRISE: "False"
+    FEATURE_ENABLE_EXTERNAL_COURSE_SYNC: "False"
```
Comment (arslanashraf7) [inline src/ol_infrastructure/applications/xpro/Pulumi.applications.xpro.Production.yaml:15]: I assume this overrides the value set in __main__.

**Review by arslanashraf7 — APPROVED** [review]
👍 

We'll need to coordinate with @mitodl/devops to put the API key in the vault.

File: src/ol_infrastructure/applications/xpro/Pulumi.applications.xpro.Production.yaml (hunk 1 above)
Comment (asadali145) [reply src/ol_infrastructure/applications/xpro/Pulumi.applications.xpro.Production.yaml:15]: @mitodl/devops Can confirm. I just followed the pattern for `FEATURE_ENABLE_ENTERPRISE `.

Comment (Ardiea) [reply src/ol_infrastructure/applications/xpro/Pulumi.applications.xpro.Production.yaml:15]: This explicitly sets `FEATURE_ENABLE_ENTERPRISE` to `False` for _just_ the production environment. Does that answer the question?

**Review by Ardiea — APPROVED** [review]

**Pull request merged**